Modules/_xxsubinterpretersmodule.c

```diff
@@ -2323,6 +2353,74 @@ PyDoc_STRVAR(channel_list_all_doc,
 \n\
 Return the list of all IDs for active channels.");
 
+
+static PyObject *
+channel_list_interpreters(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"cid", "send", "recv", NULL};
+    int64_t cid;
+    _PyChannelState *chan;
+    int send = 0;           /* Send end? */
+    int recv = 0;           /* Receive end? */
+    int64_t *ids = NULL;    /* Array of interpreter IDs */
+    int64_t count = 0;      /* Number of interpreters to return */
+    PyObject *ret = NULL;   /* Python list of interpreter IDs */
+
+    if (!PyArg_ParseTupleAndKeywords(
+            args, kwds, "O&|$pp:channel_list_interpreters",
+            kwlist, channel_id_converter, &cid, &send, &recv)) {
+        return NULL;
+    }
+
+    if ((send && recv) || (!send && !recv)) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Specify exactly one of send or recv");
+        goto except;
+    }
+
+    chan = _channels_lookup(&_globals.channels, cid, NULL);
+    if (chan == NULL) {
+        goto except;
+    }
+
+    ids = _channelends_list_interpreters(chan->ends, &count, send);
+    if (ids == NULL) {
+        goto except;
+    }
+
+    ret = PyList_New((Py_ssize_t)count);
+    if (ret == NULL) {
+        goto except;
+    }
+
+    for (int64_t i=0; i < count; i++) {
+        PyInterpreterState *interp = _PyInterpreterState_LookUpID(ids[i]);
+        PyObject *id_obj = _PyInterpreterState_GetIDObject(interp);
+        if (id_obj == NULL) {
+            goto except;
+        }
+        PyList_SET_ITEM(ret, i, id_obj);
+    }
+
+    goto finally;
+
+except:
+    Py_XDECREF(ret);
+    ret = NULL;
+
+finally:
+    PyMem_Free(ids);
+    return ret;
+}
+
+PyDoc_STRVAR(channel_list_interpreters_doc,
+"channel_list_interpreters(cid, *, send=False, recv=False) -> [id]\n\
```

Comment:
As recommended above:

```suggestion
"channel_list_interpreters(cid, *, recv=True) -> [id]\n\
```

Reply:
I've gone for `channel_list_interpreters(cid, *, send)` (no default for which end) - let me know if you'd like me to change this.